src/js/message.js

```diff
@@ -0,0 +1,131 @@
+import construct from './construct-element';
+import { throwError } from './helpers';
+
+class Message {
+	/**
+ * Class constructor.
+ * @param {HTMLElement} [messageElement] - The message element in the DOM
+ * @param {Object} [options={}] - An options object for configuring the message
+ */
+	constructor(messageElement, options) {
+		this.messageElement = messageElement;
+
+		//Default options
+		const messageClass = options && options.messageClass ? options.messageClass : 'o-message';
+		const type = options && options.type ? options.type : 'alert';
+
+		this.opts = Object.assign({}, {
+			autoOpen: true,
+			messageClass,
+			type,
+			typeVariant: `${messageClass}--${type}`,
```

Comment:
Maybe pedantic but perhaps these option names could be `typeClass`, `statusClass`. It's more clear what data the options hold, is more inline with`messageClass`, and the difference between them and the raw `type`/`status` options is clearer.

(Or maybe they should be `typeClass` and `statusClass` as direct properties of oMessage (`oMessage.typeClass`), seen as they're not options but generated based on the options. )